grid2op/multi_agent/multiAgentEnv.py

```diff
@@ -0,0 +1,557 @@
+            )
+        )
+        self.agent_order = AgentSelector(self.agents, agent_order_fn)
+        self.action_spaces = dict()
+        self.observation_spaces = dict()
+        self._build_subgrids()
+        self._build_action_spaces()
+        self._build_observation_spaces()
+        
+        
+    def reset(self) -> MADict:
+        self._cent_env.reset()
+        self._update_observations(_update_state=False)
+        return self.observations
+    
+    def step(self, action : ActionProfile) -> Tuple[MADict, MADict, MADict, MADict]:
+        raise NotImplementedError()
+    
+    def _build_subgrids(self):
+        
+        self._subgrids_cls = {
+            'action' : dict(),
+            'observation' : dict()
+        }
+        for agent_nm in self.agents : 
+            # action space
+            self._build_agent_domain(agent_nm, self._action_domains[agent_nm])
+            subgridcls = self._build_subgrid_cls_from_domain(self._action_domains[agent_nm])
+            self._subgrids_cls['action'][agent_nm] = subgridcls 
+            
+            # observation space
+            if self._observation_domains is not None:
+                self._build_agent_domain(agent_nm, self._observation_domains[agent_nm])
+                subgridcls = self._build_subgrid_cls_from_domain(self._observation_domains[agent_nm])
+                self._subgrids_cls['observation'][agent_nm] = subgridcls
+        
+    def _build_agent_domain(self, agent_nm, domain):
+
+        is_sub_in = np.full(self._cent_env.n_sub, fill_value=False, 
+                            dtype=dt_bool)
+        is_sub_in[domain['sub_id']] = True
+        domain['mask_sub'] = is_sub_in
+        
+        domain['mask_load'] = np.isin(
+            self._cent_env.load_to_subid, domain['sub_id'] 
+        )
+        domain['mask_gen'] = np.isin(
+            self._cent_env.gen_to_subid, domain['sub_id'] 
+        )
+        domain['mask_storage'] = np.isin(
+            self._cent_env.storage_to_subid, domain['sub_id'] 
+        )
+        domain['mask_shunt'] = np.isin(
+            self._cent_env.shunt_to_subid, domain['sub_id'] 
+        )
+        domain['mask_line_or'] = np.isin(
+            self._cent_env.line_or_to_subid, domain['sub_id']
+        )
+        domain['mask_line_ex'] = np.isin(
+            self._cent_env.line_ex_to_subid, domain['sub_id'] 
+        )
+        domain['mask_interco'] = domain['mask_line_or'] ^ domain['mask_line_ex']
+        domain['interco_is_origin'] = domain['mask_line_or'][domain['mask_interco']]
+        domain['mask_line_or'] = domain['mask_line_or'] & domain['mask_line_ex']
+        domain['mask_line_ex'] = domain['mask_line_or'].copy()
+        domain["agent_name"] = agent_nm
+    
+    def _relabel_subid(self, mask, new_label, id_full_grid):
+        tmp_ = id_full_grid[mask]
+        return new_label[tmp_]
+    
+    def seed(self, seed):
+        self._cent_env.seed(seed)
+        return super().seed(seed)
```

Comment:
WRONG 

1. You should not have changed this, I specifically said "don't touch anything"
2. you removed the TODO, that specfie it was wrong
3. it's wrong...

Reset it to a "normal" state, with a NotImplementedError()

Reply:
Done